tests/ClientBuilderTest.php

```diff
@@ -210,6 +210,38 @@ public function optionsDataProvider()
             ['setErrorTypes', 0],
         ];
     }
+
+    public function testCompressionPluginAdded()
+    {
+        $builder = ClientBuilder::create(['encoding' => 'gzip', 'dsn' => 'https://completelyrandom@dsn.asdf/42']);
+        $builder->getClient();
+        $reflectionProperty = new \ReflectionProperty(ClientBuilder::class, 'httpClientPlugins');
```

Comment:
I don't like testing this with reflection, but I see that there isn't another pretty way to do it.
Let's hope to be able to use a stub server ASAP for this kind of stuff.

Reply:
Instead of manually using the `ReflectionProperty` class we could simplify a bit the test by using the helper method of PHPUnit `getObjectAttribute`